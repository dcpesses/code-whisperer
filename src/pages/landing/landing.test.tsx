import {render} from '@testing-library/react';
import Landing from './index';

describe('Landing', () => {
  test('Should render as expected', () => {
    const {container} = render(<Landing />);
    expect(container).toMatchSnapshot();
  });
});
