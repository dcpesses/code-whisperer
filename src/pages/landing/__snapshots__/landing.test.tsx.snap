// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Landing > Should render as expected 1`] = `
<div>
  <div
    class="landing container-fluid"
  >
    <a
      aria-label="View source on GitHub"
      class="github-corner"
      href="https://github.com/dcpesses/code-whisperer"
    >
      <svg
        aria-hidden="true"
        height="80"
        style="fill: #fff; color: #151513; position: absolute; top: 0px; border: 0px; right: 0px;"
        viewBox="0 0 250 250"
        width="80"
      >
        <path
          d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"
        />
        <path
          class="octo-arm"
          d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2"
          fill="currentColor"
          style="transform-origin: 130px 106px;"
        />
        <path
          class="octo-body"
          d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z"
          fill="currentColor"
        />
      </svg>
    </a>
    <div
      class="container-fluid d-flex flex-column"
      id="login-screen"
    >
      <div
        class="row row-cols-auto align-items-center justify-content-center"
        id="landing"
      >
        <div
          class="col-8 col-md-6 text-center align-self-end align-self-md-center"
        >
          <div
            class="text-center pb-1"
          >
            <div>
              <img
                alt="Logo"
                class="login-logo w-50 mx-auto mb-3"
                src="/code-whisperer/src/assets/new-logo.svg"
              />
            </div>
            <h2
              class="text-center mb-0"
            >
              Code Whisperer
            </h2>
            <div
              class="text-center fs-6"
            >
              <small>
                v0.7.0
              </small>
            </div>
          </div>
        </div>
        <div
          class="col-8 col-md-6 text-center align-self-start align-self-md-center"
        >
          <a
            class="btn btn-sm fs-2 py-2 px-3 rounded-4 bg-gradient"
            href="https://id.twitch.tv/oauth2/authorize?client_id=rxkepwcnejign8h6lil4l4etlfc4kw&response_type=code&scope=chat:read chat:edit moderation:read user:manage:whispers moderator:read:chatters user:read:moderated_channels channel:read:vips channel:read:editors moderator:manage:announcements user:read:subscriptions&redirect_uri=http%3A%2F%2Flocalhost%3A5173%2Fcode-whisperer%2F%23"
          >
            Log In With 
            <strong>
              Twitch
            </strong>
          </a>
          <div
            class="text-center fs-6"
          >
            <button
              class="btn btn-sm btn-link link-secondary text-decoration-none"
            >
              Having issues? 
              <span
                class="with-checkmark"
              >
                Reset application data
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="landing-wave-wrapper"
    >
      <div
        class="landing-wave"
      >
        <section>
          <div
            class="wave wave1"
          />
          <div
            class="wave wave2"
          />
          <div
            class="wave wave3"
          />
          <div
            class="wave wave4"
          />
        </section>
      </div>
    </div>
    <div
      class="container-fluid app-promo"
    >
      <div
        class="row align-items-start justify-content-center mb-4"
      >
        <div
          class="col-sm-10 max-w-md-340"
        >
          <div
            class="pb-1"
          >
            <h1
              class="fw-bold text-center text-md-start"
            >
              <span
                class="d-inline-block"
              >
                Play With Friends,
              </span>
               
              <span
                class="d-inline-block"
              >
                Not Your Foes!
              </span>
            </h1>
            <div
              class="fs-5 my-4 text-center text-md-start"
            >
              Developed as a way to help streamers thwart Jackbox trolls, this tool gives Twitch streamers the power to decide who can play in games they host by sending invites only to approved users.
            </div>
            <h3>
              How It Works:
            </h3>
            <ul
              class="ps-4 fs-6 my-4"
            >
              <li
                class="mb-1"
              >
                Followers in chat who type 
                <b>
                  !join
                </b>
                 get added to the 
                <i>
                  Interested
                </i>
                 lobby.
              </li>
              <li
                class="mb-1"
              >
                The streamer then adds any or all of them to the 
                <i>
                  Playing
                </i>
                 section. You can even randomize who gets added too!
              </li>
              <li
                class="mb-1"
              >
                Once you enter the room code, press 
                <i>
                  Send to Queue
                </i>
                 to whisper everyone selected to play.
              </li>
            </ul>
          </div>
        </div>
        <div
          class="max-w-340 text-center"
        >
          <div
            class="twitch-chat-screenshot text-right"
          >
            <img
              alt="Twitch Chat"
              class="twitch-chat img-fluid shadow border border-secondary rounded-4 mx-auto mb-3"
              src="/code-whisperer/src/assets/twitch-chat.png"
            />
          </div>
        </div>
      </div>
      <div
        class="col-md-auto text-center"
      >
        <img
          alt="App Screenshot"
          class="screenshot img-fluid shadow border border-secondary roounded-4 mx-auto mb-3"
          src="/code-whisperer/src/assets/screenshot2.png"
        />
      </div>
      <div
        class="col-md-auto text-center"
      >
        <div
          class="fs-6 my-3"
        >
          (Oh, and did we mention it's free to use?*)
        </div>
        <a
          class="btn btn-sm fs-4 py-2 px-3 rounded-4 my-4 btn-purple bg-gradient"
          href="https://id.twitch.tv/oauth2/authorize?client_id=rxkepwcnejign8h6lil4l4etlfc4kw&response_type=code&scope=chat:read chat:edit moderation:read user:manage:whispers moderator:read:chatters user:read:moderated_channels channel:read:vips channel:read:editors moderator:manage:announcements user:read:subscriptions&redirect_uri=http%3A%2F%2Flocalhost%3A5173%2Fcode-whisperer%2F%23"
        >
          <span
            class="d-inline-block"
          >
            Go ahead, log in with 
            <strong>
              Twitch
            </strong>
          </span>
           
          <span
            class="d-inline-block"
          >
            and take it for a spin!
          </span>
        </a>
      </div>
      <div
        class="col-md-8 m-auto text-center"
      >
        <div
          class="fs-5 my-4"
        >
          View 
          <a
            class="emphasis"
            href="https://github.com/dcpesses/code-whisperer"
            rel="noreferrer"
            target="_blank"
          >
            Code Whisperer on GitHub
          </a>
           for more information, including documentation, planned features, technical support, and ways to contribute.
        </div>
      </div>
      <div
        class="col-md-auto text-center"
      >
        <div
          class="fs-6 my-4"
        >
          *Yes seriously, it is free. That said, 
          <span
            class="d-inline-block"
          >
            you're also welcome to 
            <a
              class="emphasis"
              href="https://ko-fi.com/V7V6VSUT1"
              rel="noreferrer"
              target="_blank"
            >
              leave a tip!
            </a>
          </span>
          <div
            class="py-2"
          >
            <a
              class="emphasis"
              href="https://ko-fi.com/V7V6VSUT1"
              rel="noreferrer"
              target="_blank"
            >
              <img
                class="img-fluid"
                src="https://camo.githubusercontent.com/70e2ef5e0263b261f9a2a314bb1d6919d1d43292eed117fe8fc766a68c7d96ea/68747470733a2f2f6b6f2d66692e636f6d2f696d672f676974687562627574746f6e5f736d2e737667"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
